### PR TITLE
[fix,refactor]: fix #6, fixed a lot of bugs, optimized several things

### LINKED ISSUE #6: HTTP/1.0 is deprecated
Great job with this repo, looks amazing. Only one thing stood out to me, that it's not using HTTP/1.1 which every non http-2 client should be using instead of 1.0 that's been deprecated for like a decade

https://github.com/top-gg/rust-sdk/blob/27de52628e6c7c1890be84fccac734f7451040f9/src/http.rs#L46

### PR DESCRIPTION
**Pull request by null8626**
The following pull request ACTUALLY fixes #6 (apparently #8 didn't do anything, i didn't double-check)

Other than that, as i've learned from Mac in #8, i've did several changes to the library too, such as bug-fixes and lot of improvements/optimizations. (Bumped to version `1.1.0`)

### CODE REVIEW

**Review by Anonymous4078 — APPROVED** [review]

**Comment by Luke-6723** [comment]
bets on it being a month before merge lol

**Comment by null8626** [comment]
> bets on it being a month before merge lol

at this point just make me maintainer lmao

i have the full access of publishing the library into crates.io anyway - and considering my past pull requests, i should be trustworthy enough

**Comment by pixxies** [comment]
I would if I could but I don't have access to do that unfortunately, and getting the person who can do it to do it for me will take some time.

**Pull request merged**
